lib/Module/IntrinsicCleaner.cpp

```diff
@@ -49,6 +49,58 @@ bool IntrinsicCleanerPass::runOnModule(Module &M) {
   return dirty;
 }
 
+llvm::Constant* IntrinsicCleanerPass::createConstantVector(llvm::LLVMContext& ctx, 
+  llvm::Value* pConstantVector, unsigned mode) { 
```

Comment:
```suggestion
  llvm::Value* pConstantVector, enum Mode mode) { 
```

Replace the `unsigned` with an `enum` defined by you (Look at the great `color` example from https://en.cppreference.com/w/cpp/language/enum). This is much more expressive and simplifies the code a lot.

Reply:
It should be changed like this